activation/malfeasance2.go

```diff
@@ -38,6 +38,14 @@ func (mh *MalfeasanceHandlerV2) Info(data []byte) (map[string]string, error) {
 	return info, nil
 }
 
+func (mh *MalfeasanceHandlerV2) ReportLabels(data []byte) []string {
+	proof, err := mh.decodeProof(data)
+	if err != nil {
+		return []string{"ATX", "unknown"}
+	}
+	return []string{"ATX", proof.String()}
```

Comment:
It's confusing that a `.String()` is used here. AFAIR, the implementation returns the type of the proof, but to the reader of the code it could be anything. How about changing the `Proof interface` to have a method `Type() string` instead?

Reply:
The `wire.Proof` interface already has a `Type` method that is used to distinguish different proof types during en-/decoding. Arguably you are right that `String` should print a string representation of the specific object, not just the name of the type. I'll rename the method to `Name`. 🙂